### ISSUE TITLE
Update tough-cookie to v5

### ISSUE THREAD

**Issue by wojtekmaj** [bundled-es-modules/tough-cookie#1]
v5 has been released:

https://github.com/salesforce/tough-cookie/releases/tag/v5.0.0

- Removed psl dependency, removing Node.js 22 deprecation warnings
- Rewritten in TypeScript, so no @types/tough-cookie needed anymore

**Comment by yap-paf** [comment]
Yes, please update @kettanaito  , it will remove `DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.`

**Comment by kettanaito** [comment]
@yap-paf, the PR is open at https://github.com/mswjs/msw/pull/2356. Needs some TS refactoring, new tough-cookie has breaking changes in types from what I can see.

**Comment by kettanaito** [comment]
Also, PRs to this mirror are still welcome!

**Comment by stramel** [comment]
Added a PR to bump this mirror to v5.

**Comment by roll** [comment]
Hi, can you please take a look at this? It seems to be causing `vitest` to throw `punycode` deprecation warning
